src/structures/interfaces/TextBasedChannel.js

```diff
@@ -159,25 +159,42 @@ class TextBasedChannel {
   /**
    * Starts a typing indicator in the channel.
    * @param {number} [count] The number of times startTyping should be considered to have been called
+   * @returns {Promise}
    * @example
    * // Start typing in a channel
    * channel.startTyping();
    */
   startTyping(count) {
     if (typeof count !== 'undefined' && count < 1) throw new RangeError('TYPING_COUNT');
     if (!this.client.user._typing.has(this.id)) {
+      let resolve;
+      let reject;
+      const promise = new Promise((res, rej) => { [resolve, reject] = [res, rej]; });
```

Comment:
this is a huge evil anti-pattern, just wrap the following code in a promise and set it in the entry or return it